html/Callbacks/RT-Extension-TicketActions/Elements/Tabs/Privileged

```diff
@@ -22,13 +24,50 @@
 					);
 				}
 				if ($Ticket->Status ne 'stalled') {
+					my $holidaysRepo;
+					my %holidays = ();
+					if (defined $holidaysCountry) {
+						eval 'use Date::Holidays;';
+						$holidaysRepo = Date::Holidays->new(countrycode => $holidaysCountry);
+					}
+
 					foreach my $fu_day (@follow_up_days) {
 						my $date = RT::Date->new($session{'CurrentUser'});
 						$date->SetToNow();
 						$date->Set(Format => 'iso', Value => $date->Date() . ' ' . $follow_up_time);
 						$date->AddDays($fu_day);
+
+						if (defined $holidaysRepo) {
+							while ($date->ISO() =~ /^(\d+)-(\d+-\d+)/) {
+								my ($year, $day) = ($1, $2);
+
+								if (! exists $holidays{$year}) {
+									$holidays{$year} = {};
+
+									for my $holiday (@{$holidaysRepo->holidays(year => $year)}) {
+										my $holiDate = RT::Date->new($session{'CurrentUser'});
+										$holiDate->Unix($holiday);
+
+										if ($holiDate->ISO() =~ /^\d+-(\d+-\d+)/) {
+											$holidays{$year}->{$1} = 1
+										}
+									}
+								}
+
+								if (exists $skipDow{(localtime $date->Unix())[6]}
```

Comment:
Please add a comment what exactly is done here, or make it more simple with a local variable.

Reply:
Done.